presto-main/src/main/java/io/prestosql/execution/resourcegroups/InternalResourceGroup.java

```diff
@@ -669,24 +668,34 @@ private void startInBackground(ManagedQueryExecution query)
     private void queryFinished(ManagedQueryExecution query)
     {
         synchronized (root) {
-            if (!runningQueries.contains(query) && !queuedQueries.contains(query)) {
+            if (!runningQueries.keySet().contains(query) && !queuedQueries.contains(query)) {
                 // Query has already been cleaned up
                 return;
             }
-            // Only count the CPU time if the query succeeded, or the failure was the fault of the user
-            if (!query.getErrorCode().isPresent() || query.getErrorCode().get().getType() == USER_ERROR) {
-                InternalResourceGroup group = this;
-                while (group != null) {
-                    group.cpuUsageMillis = saturatedAdd(group.cpuUsageMillis, query.getTotalCpuTime().toMillis());
-                    group = group.parent.orElse(null);
-                }
-            }
-            if (runningQueries.contains(query)) {
+
+            if (runningQueries.keySet().contains(query)) {
```

Comment:
Why are we no longer checking for USER_ERROR/success? Should we allow system failures to impact user-available CPU?

Reply:
In the current implementation, the user gets punished for memory usage even if the query fails due to a system error.  

Also, now since cpu usage is updated on the fly, user might already start getting punished before the query fails. 

That is why I felt that we should not discriminate between success/failure here. What do you think?